### PR TITLE
ci(sdk): add package.json to sdk exports block

### PR DESCRIPTION
**Pull request by heypoom**
Adds `"./package.json": "./package.json"` mapping to the `exports` block of the SDK. This helps module resolution to work better in cross-version end-to-end tests (based on @deniskaber's suggestions)

### CODE REVIEW

**Review by deniskaber — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/package.template.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
       "types": "./dist/nextjs.d.ts",
       "require": "./dist/nextjs.cjs",
       "import": "./dist/nextjs.js"
-    }
+    },
+    "./package.json": "./package.json"
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/package.template.json:21]: Will this make it show up in the autocomplete when importing from the host app?

Does `resolve(`@metabase/embedding-sdk-react`) resolve to the root folder? if so I think we can just manually calculate the path to the package.json from that maybe?

Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/package.template.json:21]: @npretto resolve was resolving to "index.ts" in the source folder (enterprise/frontend/src/embedding-sdk), weirdly enough, so in https://github.com/metabase/metabase/pull/51268 I manually resolved to node_modules in the component tests.

Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/package.template.json:21]: > Will this make it show up in the autocomplete when importing from the host app?

Good question, let me try to build this locally and see if it affects imports.

**Review by npretto — APPROVED** [review]

**Comment by heypoom** [comment]
note: adding temporary commit https://github.com/metabase/metabase/pull/51328/commits/ad95e1f6d9411639330a03611445ba5eb9c3cc25 that restores the disabled workflows, and makes it run against this branch. I'll revert that commit once the CI checks in the commit passed.

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**